toolchain/sem_ir/name_scope.h

```diff
@@ -18,13 +18,104 @@ enum class AccessKind : int8_t {
   Private,
 };
 
+// Represents the result of a name lookup.
+// A lookup result can be:
+// * Successful returning a valid `InstId`,
+// * Not found when the name wasn't declared or poisoned.
+// * Poisoned when the name wasn't declared but was poisoned.
+// * Represent that an error has occurred during lookup. This is still
+// considered found and the error `InstId` is considered valid.
+class ScopeLookupResult {
+ public:
+  // Creates a non poisoned result.
+  // Depending on the given `inst_id`, it can be:
+  // * Found: Lookup was successful and `inst_id` is valid. If `inst_id` is
+  // known to be valid, use `CreateFound()`.
+  // * Not Found: Lookup returned no results and `inst_id` is invalid. If
+  // `inst_id` is known to be invalid, use `CreateNotFound()`.
+  // * Error: Lookup encountered an error and `inst_id` is
+  // `ErrorInst::SingletonInstId`, which is considered valid and found. If
+  // `inst_id` is known to be `ErrorInst::SingletonInstId`, `use CreateError()`.
+  //
+  // To create a poisoned result, use `CreatePoisoned()`.
+  static auto Create(InstId inst_id, AccessKind access_kind)
+      -> ScopeLookupResult {
+    return ScopeLookupResult(inst_id, access_kind, false);
```

Comment:
```suggestion
    return ScopeLookupResult(inst_id, access_kind, /*is_poisoned=*/false);
```

Generally I think it's helpful to mark the parameter name when a common literal value (false, true, nullptr, etc) are directly passed.

Reply:
Done.